kits/apptentive-kit/src/main/java/com/mparticle/kits/ApptentiveKit.java

```diff
@@ -0,0 +1,250 @@
+package com.mparticle.kits;
+
+import android.app.Activity;
+import android.app.Application;
+import android.location.Location;
+import android.text.TextUtils;
+
+import com.apptentive.android.sdk.Apptentive;
+import com.apptentive.android.sdk.ApptentiveInternal;
+import com.apptentive.android.sdk.lifecycle.ApptentiveActivityLifecycleCallbacks;
+import com.apptentive.android.sdk.model.CommerceExtendedData;
+import com.mparticle.MPEvent;
+import com.mparticle.MParticle;
+import com.mparticle.commerce.CommerceEvent;
+import com.mparticle.commerce.Product;
+import com.mparticle.commerce.TransactionAttributes;
+import com.mparticle.internal.CommerceEventUtil;
+import com.mparticle.internal.ConfigManager;
+
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class ApptentiveKit extends AbstractKit implements ClientSideForwarder, ECommerceForwarder, ActivityLifecycleForwarder {
+	private static final String API_KEY = "appKey";
+	private static final String CUSTOM_DIMENSIONS = "customDimensions";
+	private static final String RAW_LTV = "trackClvAsRawValue";
+	private JSONArray customDimensionJson = null;
+	private boolean trackAsRawLtv = false;
+	private ApptentiveActivityLifecycleCallbacks callbacks;
+
+	@Override
+	protected AbstractKit update() {
+		if (callbacks == null) {
+			callbacks = new ApptentiveActivityLifecycleCallbacks(context.getApplicationContext());
+		}
+		/* Note mParticle will delegate lifecycle management to the above callbacks. No need to register ApptentiveActivityLifecycleCallbacks
+         * through Apptentive. But do need to initialize Apptentive
+         *
+         */
+		ApptentiveInternal.createInstance(context.getApplicationContext(), properties.get(API_KEY));
+
+		return this;
+
+	}
+
+	@Override
+	public Object getInstance(Activity activity) {
+		return null;
+	}
+
+	@Override
+	public String getName() {
+		return "Apptentive";
+	}
+
+	@Override
+	public boolean isOriginator(String uri) {
+		return uri != null && uri.toLowerCase().contains("Apptentive.com");
+	}
+
+
+	@Override
+	public void setUserIdentity(String id, MParticle.IdentityType identityType) {
+		if (identityType.equals(MParticle.IdentityType.Email)) {
+			Apptentive.setPersonEmail(id);
+		} else if (identityType.equals(MParticle.IdentityType.CustomerId)) {
+			Apptentive.setPersonName(id);
+		}
+	}
+
+	@Override
+	void setUserAttributes(JSONObject mUserAttributes) {
+		HashSet<String> attributeKeys = null;
+		if (customDimensionJson != null) {
```

Comment:
hey @barry-li - custom dimensions are a feature specific to the Localytics Kit. For Apptentive, all that you need to do here is iterate over the `mUserAttributes` object and call `Apptentive.addCustomPersonData()` for each entry:

```
Iterator<String> keys = mUserAttributes.keys();
while (keys.hasNext()) {
    String key = keys.next();
    String value = mUserAttributes.optString(key, "");
    Apptentive.addCustomPersonData(attributeName, attributeValue);
}
```